Release summary for 8.19 (#18473)
coq/coq: doc/sphinx/changes.rst

  - New command :cmd:`Attributes` to assign attributes such as
    :attr:`deprecated` to a library file.

Notable breaking changes:

  - :tacn:`replace` with `by tac` does not automatically attempt to solve
    the generated equality subgoal using the hypotheses.
    Use `by first [assumption | symmetry;assumption | tac]`
    if you need the previous behaviour.

  - :ref:`Removed old deprecated files <819Stdlib>` from the standard library.

See the `Changes in 8.19.0`_ section below for the detailed list of changes,
including potentially breaking changes marked with **Changed**.
Coq's `reference manual for 8.19 <https://coq.github.io/doc/v8.19/refman>`_,
`documentation of the 8.19 standard library <https://coq.github.io/doc/v8.19/stdlib>`_
and `developer documentation of the 8.19 ML API <https://coq.github.io/doc/v8.19/api>`_
are also available.

Maxime Dénès and Thierry Martinez moved the CI away from `gitlab.com <http://gitlab.com>`_ to
use Inria supported runner machines through  `gitlab.inria.fr <https://gitlab.inria.fr>`_.

Théo Zimmermann maintained `coqbot <https://github.com/coq/bot>`_ used
to run Coq's CI and other pull request management tasks.

> 

Reviewer: I CC @Zimmi48 for review of this item and the CI migration item, in case some contributors are missing.

Reviewer 2: Indeed, to be more exhaustive, we could add "with support from Erik Martin-Dorel and Théo Zimmermann" to the CI migration part, and "with help from Ali Caglayan and Jason Gross" to the coqbot maintenance part.

> Maxime Dénès and Thierry Martinez moved the CI away from `gitlab.com <http://gitlab.com>` to
> use Inria supported runner machines through  `gitlab.inria.fr <https://gitlab.inria.fr>`.
> 
> Théo Zimmermann maintained `coqbot <https://github.com/coq/bot>` used
> to run Coq's CI and other pull request management tasks.
> 
> Jason Gross maintained the `bug minimizer <https://github.com/JasonGross/coq-tools>`
> and its `automatic use through coqbot <https://github.com/coq/coq/wiki/Coqbot-minimize-feature>`.

Reviewer 2: ```suggestion
Maxime Dénès and Thierry Martinez moved the CI away from `gitlab.com <http://gitlab.com>`_ to
use Inria supported runner machines through  `gitlab.inria.fr <https://gitlab.inria.fr>`_.

Théo Zimmermann maintained `coqbot <https://github.com/coq/bot>`_ used
to run Coq's CI and other pull request management tasks.

Jason Gross maintained the `bug minimizer <https://github.com/JasonGross/coq-tools>`_
and its `automatic use through coqbot <https://github.com/coq/coq/wiki/Coqbot-minimize-feature>`_.
```

Jaime Arias and Erik Martin-Dorel have maintained the `Coq Docker images
<https://hub.docker.com/r/coqorg/coq>`_ that are used in many Coq
projects for continuous integration.

Maxime Dénès, Paolo G. Giarrusso, Huỳnh Trần Khanh, Romain Tetley and Laurent Théry have
maintained the VsCoq extension for VS Code.

> 

Reviewer: What's the criteria for mentioning projects in this list? Should be Dune, jsCoq, SerAPI, coq-lsp mentioned?

Author: Not sure TBH
I think we want work which supports the development of Coq, not work aimed at users (that would go more in coq platform credits or individual project credits)
by that logic I think we would include dune and remove vscoq

Reviewer 3: From that perspective, the connection between the Platform and VsCoq is very weak. From my perspective, the inclusion of VsCoq makes sense due to it being a long-term Coq Consortium project (like what the Coq Platform now is becoming, in effect).

Author: @ejgallego not sure which projects will end up included but which people would you mention for the projects you suggested?

Reviewer: Off the top of my head (I can verify this later)

- jsCoq is Shachar Itzhaky and me (plus some contributors)
- Dune is Ali Caglayan, Rudi Gringerb, and me (and actually Rodolphe Lepigre did important contribs)
- coq-lsp is Ali, Bhakti Shah, me (and maybe some other important contribs should be mention)
- serApi is me, Karl, and Coq team (due to many overlays)

Author: some discussion is happening on zulip
https://coq.zulipchat.com/#narrow/stream/237656-Coq-devs-.26-plugin-devs/topic/credits.20for.208.2E19

> used by Coq projects for continuous integration.

Author: new
This reads a bit awkward IMO to have 
~~~
XX maintained the docker images used in Coq projects for CI.

YY maintained the nix toolbox used in Coq projects for CI.
~~~
maybe we should combine the paragraphs? eg
~~~
XX maintained the docker images and YY maintained the nix toolbox used in Coq projects for CI.
~~~
?

> The opam repository for Coq packages has been maintained by
> Guillaume Claret, Karl Palmskog, Matthieu Sozeau and Enrico Tassi with
> contributions from many users. A list of packages is available at
> https://coq.inria.fr/opam/www/.

Reviewer 3: ```suggestion
The opam repository for Coq packages has been maintained by
Guillaume Claret, Guillaume Melquiond, Karl Palmskog and Enrico Tassi with
contributions from many users. A list of packages is `available on the Coq website <https://coq.inria.fr/coq-package-index>`_.
```
I think it's accurate to say Matthieu is working on other things these days while Guillaume Melquiond is actively merging PRs. I change the URL because Maxime reorganized the website (old link is forwarded to the link given here).

> used to build Coq itself and many Coq projects.

Author: new

Reviewer: Thanks @SkySkimmer , LGTM, note a typo on Rudi's surname, it is "Rudi Grinberg"

The `Coq Platform <https://github.com/coq/platform>`_ has been maintained
by Michael Soegtrop, with help from Karl Palmskog, Pierre Roux, Enrico Tassi and

> Théo Zimmermann.

Reviewer 2: Nowadays, Romain Tetley should be listed there as well.

Our current maintainers are Yves Bertot, Frédéric Besson, Ana Borges,
Ali Caglayan, Tej Chajed, Cyril Cohen, Pierre Corbineau, Pierre
Courtieu, Andres Erbsen, Jim Fehrle, Julien Forest, Emilio Jesús
Gallego Arias, Gaëtan Gilbert, Georges Gonthier, Benjamin Grégoire,
Jason Gross, Hugo Herbelin, Vincent Laporte, Olivier Laurent, Assia
Mahboubi, Kenji Maillard, Guillaume Melquiond, Pierre-Marie Pédrot,
Clément Pit-Claudel, Pierre Roux, Kazuhiko Sakaguchi, Vincent Semeria,
Michael Soegtrop, Arnaud Spiwack, Matthieu Sozeau, Enrico Tassi,
Laurent Théry, Anton Trunov, Li-yao Xia and Théo Zimmermann. See the
`Coq Team face book <https://coq.inria.fr/coq-team.html>`_ page for

> more details.

Author: I removed @maximedenes from the 8.17 list, I don't think we have new maintainers since then?

Reviewer 4: Maybe @Villetaneuse (Pierre Rousselin).

Reviewer 5: I'm not a maintainer. Maybe later when/if I feel competent enough.

(TODO resolve usernames)
The 40 contributors to the 8.19 version are:

> quarkcool, soukouki, Khalid Abdullah, Tanaka Akira, Isaac van Bakel,

Author: @quarkcool @soukouki if you want your name to be used in this list instead of your github username please say what your name is.

Reviewer 6: Please use the name "Sotaro Okada".

including potentially breaking changes marked with **Changed**.
Coq's `reference manual for 8.19 <https://coq.github.io/doc/v8.19/refman>`_,
`documentation of the 8.19 standard library <https://coq.github.io/doc/v8.19/stdlib>`_
and `developer documentation of the 8.19 ML API <https://coq.github.io/doc/v8.19/api>`_
are also available.

CREDITS TODO

(TODO update, this is copied from 8.17)
Our current maintainers are Yves Bertot, Frédéric Besson, Ana Borges,
Ali Caglayan, Tej Chajed, Cyril Cohen, Pierre Corbineau, Pierre Courtieu, Maxime Dénès,
Andres Erbsen, Jim Fehrle, Julien Forest, Emilio Jesús Gallego Arias, Gaëtan Gilbert,
Georges Gonthier, Benjamin Grégoire, Jason Gross, Hugo Herbelin,
Vincent Laporte, Olivier Laurent, Assia Mahboubi, Kenji Maillard,
Guillaume Melquiond, Pierre-Marie Pédrot, Clément Pit-Claudel, Pierre Roux,
Kazuhiko Sakaguchi, Vincent Semeria, Michael Soegtrop, Arnaud Spiwack,
Matthieu Sozeau, Enrico Tassi, Laurent Théry, Anton Trunov, Li-yao Xia
and Théo Zimmermann. See the `Coq Team face book <https://coq.inria.fr/coq-team.html>`_
page for more details.

(TODO resolve usernames)
The 40 contributors to the 8.19 version are:

> quarkcool, soukouki, Khalid Abdullah, Tanaka Akira,

Author: @quarkcool @soukouki if you want your name to be used in this list instead of your github username please say what your name is.
